src/Android/HaskellActivity.hsc

```diff
@@ -118,9 +122,14 @@ activityCallbacksToPtrs ac = ActivityCallbacksPtrs
   <*> wrapIO (_activityCallbacks_onRestart ac)
   <*> wrapIO (_activityCallbacks_onBackPressed ac)
   <*> wrapCStringCStringIO (\a b -> do
-                               a' <- peekCString a
-                               b' <- peekCString b
-                               _activityCallbacks_onNewIntent ac a' b')
+        a' <- peekCString a
+        b' <- peekCString b
+        _activityCallbacks_onNewIntent ac a' b'
+      )
+  <*> wrapCStringIO (\token -> do
+        token' <- peekCString token
```

Comment:
In original code this did `T.decodeUtf8 . packCString`. I'm a bit worried about utf8 here.